### PR TITLE
feat: add flake8 diagnostics

### PR DESCRIPTION
**Pull request by yujinyuz**
[asciicast](https://asciinema.org/a/ASx3KjSZVoKoOBMMjttCk0XJo)

### CODE REVIEW

**Comment by yujinyuz** [comment]
I was checking other diagnostics and I'm not quite sure about the purpose of what this was doing

https://github.com/jose-elias-alvarez/null-ls.nvim/blob/ec58d64d49fcea19447d1e17ce6be33908eb219d/lua/null-ls/builtins/diagnostics.lua#L25-L33

Doesn't it already return the row and column?

```
"was stolen" may be passive voice on line 4 at column 11
```

**Comment by jose-elias-alvarez** [comment]
Looks good! I don't use Python much, but the first element is an error code, right? If so, we should probably extract it from the string and insert it as `diagnostic.code`. 

No idea about the `write-good` thing (I wrote those mostly as proofs-of-concept, though they work on my end) but if you can find a way to improve an existing built-in, feel free to put in a PR for that, too.

**Comment by yujinyuz** [comment]
@jose-elias-alvarez Here's what it returns

```
users/models.py:10:24: E202 whitespace before ')'
```

Only `10:24 and E202 whitespace before ')'` is extracted.

I'm not sure what you mean by inserting it to `diagnostic.code`.

Do you have samples I can use for reference?

**Comment by jose-elias-alvarez** [comment]
If you look at how [ESLint](https://eslint.org/docs/user-guide/formatters/#table) formats errors, I think the `Rule ID` column of the table corresponds to `E202` in your example. [Here](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/56d4b76203d2b442353e25c247f49fa5ca70f42e/lua/null-ls/builtins/diagnostics.lua#L248), you can see we insert the rule ID as `diagnostic.code` (the example is a bit different because ESLint outputs to JSON, but the idea is the same). 

For consistency, I think we should extract `E202` from the message and assign it to `code`, which will make it display nicely with integrations like [trouble.nvim](https://github.com/folke/trouble.nvim).

File: lua/null-ls/builtins/diagnostics.lua [hunk 1]
```diff
@@ -302,4 +302,47 @@ M.hadolint = h.make_builtin({
...
+        command = "flake8",
+        to_stdin = true,
+        to_stderr = true,
+        args = { "--stdin-display-name", "$FILENAME", "-" },
+        format = "line",
+        check_exit_code = function(code)
+            return code == 0 or code == 255
+        end,
+        on_output = function(line, params)
+            local pos = vim.split(string.match(line, "%d+:%d+:.+"), ":")
```
Comment (stephan-hesselmann-by) [inline lua/null-ls/builtins/diagnostics.lua:319]: You can just do
```
local row, col, msg = line:match(":(%d+):(%d+): (.*)")
```

**Comment by yujinyuz** [comment]
It works great for me

Screen Shot 2021-07-20 at 13 04 52

**Pull request merged**